### PR TITLE
Upgrade PrototypeJs from 1.7 to 1.7.3

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
Upgraded version of prototype.js library (from 1.7 to the latest 1.7.3).

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32847 : Investigate upgrade to the latest PrototypeJS

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by karyna-tsymbal-atwix** [comment]
@magento run all tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
@karyna-tsymbal-atwix, could you collect the places where PrototypeJS are used? This is a pretty old library. It's better to get rid of it completely rather than update it.

**Comment by andrewbess** [comment]
> could you collect the places where PrototypeJS are used? This is a pretty old library. It's better to get rid of it completely rather than update it.

Hello, @ihor-sviziev thank you for your suggestion.
We have already thought about this and we together with @sivaschenko was decided to update this library now. Next, we will continue work to remove prototypeJs.

**Review by ihor-sviziev — APPROVED** [review]

**Review by andrewbess — APPROVED** [review]

**Comment by andrewbess** [comment]
@magento run all tests

File: app/code/Magento/User/view/adminhtml/web/app-config.js [hunk 1]
```diff
@@ -27,7 +27,6 @@ require.config({
         'jquery/validate': 'jquery/jquery.validate',
         'jquery/hover-intent': 'jquery/jquery.hoverIntent',
         'jquery/file-uploader': 'jquery/fileUploader/jquery.fileuploader',
-        'prototype': 'prototype/prototype-amd',
```
Comment (ihor-sviziev) [inline app/code/Magento/User/view/adminhtml/web/app-config.js:30]: Could you explain, why was it removed? Tbh it's not clear why it was ever needed

Comment (karyna-t) [reply app/code/Magento/User/view/adminhtml/web/app-config.js:30]: Hello @ihor-sviziev  , 
lib/web/prototype/prototype-amd.js  is empty, and it was empty back in version 2.0.
This is the only mention of it in the current codebase, so we decided to remove it.
Maybe that .js file came from M1.

Now, after your question, I'm thinking maybe we don't need this app-config.js file at all.
https://github.com/magento/magento2/commit/12a9f66ec6dce93db33bc3e342c39b14c8d4e994
Looks like there was a task to move everything from app-config.js to requirejs-config, but the file itself wasn't deleted. What do you think?

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Functional Tests B2B

File: app/code/Magento/User/view/adminhtml/web/app-config.js (hunk 1 above)
Comment (ihor-sviziev) [reply app/code/Magento/User/view/adminhtml/web/app-config.js:30]: I think we should remove it

**Review by Den4ik — APPROVED** [review]

**Comment by Den4ik** [comment]
@magento run Functional Tests B2B

**Comment by andrewbess** [comment]
@magento run all tests

**Review by andrewbess — APPROVED** [review]

**Comment by andrewbess** [comment]
@magento run Functional Tests EE

**Comment by andrewbess** [comment]
@magento run Functional Tests B2B

**Comment by andrewbess** [comment]
@magento run Functional Tests B2B

**Comment by eduard13** [comment]
@magento run Functional Tests B2B

**Pull request merged**